### ISSUE TITLE
Provisioners started by user account do not show up in provisioners UI

### ISSUE THREAD

**Issue by bpmct** [coder/coder#14867]
To reproduce:

```sh
coder login <deployment-url> # as a user with `Owner`
coder provisioner start --org default
```

Navigate to the organization provisioners page. Nothing shows up.

IMO there are a couple reasons we may want to continue supporting a flow like this:

1. Service account/bot starting provisioners versus keys. Can have better security versus hardcoded keys as it can be OIDC/session based
2. Demos

Also, I documented this in #14778

**Comment by bpmct** [comment]
@Emyrk @jaaydenh Can we investigate getting a patch for this in before we release?

**Comment by Emyrk** [comment]
Are these user scoped provisioners?

**Comment by bpmct** [comment]
Nope. Org scoped ones but started using the users CLI auth versus a key.

**Comment by Emyrk** [comment]
@f0ssel are these keys using a type of auth we did not consider.

**Comment by f0ssel** [comment]
It's using the "user-auth" hardcoded key and being omitted like expected. These can be org scoped or user scoped. 

From my slack response-
> Yes, right now it’s a hard coded auth type “user-auth” in the database anytime a user provision connects. We would need to make some changes to how we associate a provisioner with the user and that required enough changes that during storyboarding we discussed and decided to leave user auth off the provisioners page for now until we can display it in a way that makes sense to the user. Right now the best we can do is say “heres the provisioners that used user auth, but idk what users they are”. 

Since this is evidence that maybe this behavior isn't what we want, maybe we can just display the user authed provisioners grouped by a single "user-auth" key group on the frontend without any association to the user. That's easy enough to do on the backend without needing to change our data model. What do we think about that?

**Comment by bpmct** [comment]
Ah. I'm in favor of a single `user-auth` key in the frontend. That's totally fine

**Comment by bpmct** [comment]
https://github.com/user-attachments/assets/d5fe8abc-8022-4bac-8474-c5cc8814dbba

I'm still observing this in dev.coder.com

**Comment by Kira-Pilot** [comment]
@f0ssel @aslilac can one of you look into this please?
